client/my-sites/themes/single-site.jsx

```diff
@@ -37,30 +37,31 @@ const ThemesSingleSite = ( props ) => {
 		{ analyticsPath, analyticsPageTitle, isJetpack, translate } = props,
 		jetpackEnabled = config.isEnabled( 'manage/themes-jetpack' );
 
-	if ( isJetpack && jetpackEnabled && ! site.hasJetpackThemes ) {
-		return <JetpackUpgradeMessage site={ site } />;
-	}
-
-	if ( isJetpack && jetpackEnabled && ! site.canManage() ) {
-		return <JetpackManageDisabledMessage site={ site } />;
-	}
-
-	if ( isJetpack && ! jetpackEnabled ) {
-		return (
-			<Main className="themes">
-				<PageViewTracker path={ analyticsPath } title={ analyticsPageTitle }/>
-				<SidebarNavigation />
-				<CurrentTheme
-					site={ site }
-					canCustomize={ site && site.isCustomizable() } />
-				<EmptyContent title={ translate( 'Changing Themes?' ) }
-					line={ translate( 'Use your site theme browser to manage themes.' ) }
-					action={ translate( 'Open Site Theme Browser' ) }
-					actionURL={ site.options.admin_url + 'themes.php' }
-					actionTarget="_blank"
-					illustration="/calypso/images/drake/drake-jetpack.svg" />
-			</Main>
-		);
+	if ( isJetpack ) {
```

Comment:
Hm, maybe a pattern would be something like using another functional component, which could even be defined within the same file since it wouldn't be used outside the closure.

```js
if ( isJetpack ) {
  return <JetpackThemePage site={ site } />
}
```